runtime/server/watchfiles.go

```diff
@@ -0,0 +1,115 @@
+package server
+
+import (
+	"fmt"
+	"github.com/bmatcuk/doublestar/v4"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+)
+
+const (
+	WatchFilesMethod = "WatchFiles"
+)
+
+func (s *Server) WatchFiles(req *runtimev1.WatchFilesRequest, stream runtimev1.RuntimeService_WatchFilesServer) error {
+	repo, err := s.runtime.Repo(stream.Context(), req.InstanceId)
+	if err != nil {
+		return err
+	}
+
+	if repo.Driver() != "file" {
+		return fmt.Errorf("%s repository is not supported", repo.Driver())
+	}
```

Comment:
Instead of checking the repo is a `file` type in the server, could we add something like a `Watch(ctx, callback func(event)) error` function on `drivers.RepoStore` instead? And have it return an error from any repo driver that doesn't support it (like `github`, which is readonly).

Imagine we wanted to add a DB-backed repo store in the future (to support the data modeler in the cloud). In that implementation, we could then use something like Postgres' `LISTEN/NOTIFY` to watch changes inside the driver, without refactoring the server implementation.